game-core/src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java

```diff
@@ -121,11 +117,9 @@ private HeadlessGameServer() {
     }
     lobbyWatcherResetupThread.scheduleAtFixedRate(() -> {
       try {
-        restartLobbyWatcher(setupPanelModel, game);
+        restartLobbyWatcher();
       } catch (final Exception e) {
-        Interruptibles.sleep(10 * 60 * 1000);
-        // try again, but don't catch it this time
-        restartLobbyWatcher(setupPanelModel, game);
+        log.log(Level.WARNING, "Failed to restart Lobby watcher", e);
```

Comment:
If you prefer a different behaviour, for example just propagating the error directly etc.
I'd be happy to change that.

Reply:
Oh, and as a general question, I'm not exactly sure what exactly this code does.
I think it would reconnect a bot to the lobby when a connection was lost or something, but the standard reconnection timer (172800 seconds) is a little bit too long for my understanding.

Reply:
I'm not sure if LobbyWatcher works correctly, I was looking at that the other day and did not come to a conclusion one way or another. If the server is not up to start with, the code seemed to run into some issues.. So I would agree that it does not seem like things are quite right to start with.